src/quo2/components/wallet/network_routing/animation.cljs

```diff
@@ -0,0 +1,110 @@
+          :let          [number-bars-before (+ number-previous-bars
+                                               (inc bar-idx))
+                         new-translation-x  (->> (take number-bars-before bars-widths-negative)
+                                                 (reduce +)
+                                                 (* 1.05))]]
+    (reanimated/animate (:translate-x-shared-value bar) new-translation-x pressed-bar-timing)
+    (js/setTimeout
+     #(let [translate-x-value (reanimated/get-shared-value (:translate-x-shared-value bar))
+            hidden-position   (- translate-x-value extra-offset)]
+        (reanimated/set-shared-value (:translate-x-shared-value bar) hidden-position))
+     pressed-bar-timing)))
+
+(def ^:private max-limit-bar-timing 300)
+
+(defn show-max-limit-bar
+  [max-limit-bar-opacity]
+  (reanimated/animate max-limit-bar-opacity 1 max-limit-bar-timing))
+
+(defn hide-max-limit-bar
+  [max-limit-bar-opacity]
+  (reanimated/animate max-limit-bar-opacity 0 max-limit-bar-timing))
+
+(defn reset-bars-positions
+  [bars unlock-press-fn]
+  (let [bars-reset-timing 500]
+    (doseq [{:keys [translate-x-shared-value]} bars]
+      (reanimated/animate translate-x-shared-value 0 bars-reset-timing))
+    (js/setTimeout unlock-press-fn bars-reset-timing)))
+
+(defn align-bars-off-screen
+  [{:keys [new-network-values network-bars amount->width]}]
+  (let [width-to-off-screen (->> new-network-values
+                                 (reduce #(- %1 (:amount %2)) 0)
+                                 (amount->width))]
+    (doseq [[{new-amount :amount} bar] (map vector new-network-values network-bars)]
+      (reanimated/set-shared-value (:amount-shared-value bar) new-amount)
+      (reanimated/set-shared-value (:translate-x-shared-value bar) (* 2 width-to-off-screen))
+      (js/setTimeout
+       #(reanimated/set-shared-value (:translate-x-shared-value bar) width-to-off-screen)
+       1))))
+
+(def ^:private hide-bar-timing 400)
+
+(defn hide-pressed-bar
+  [{{:keys [translate-x-shared-value
+            amount-shared-value]} :bar
+    amount->width                 :amount->width}]
+  (let [bar-width         (amount->width (reanimated/get-shared-value amount-shared-value))
+        new-translation-x (- (reanimated/get-shared-value translate-x-shared-value)
+                             bar-width)]
+    (reanimated/animate translate-x-shared-value new-translation-x hide-bar-timing)))
+
+(defn update-bar-values-and-reset-animations
+  [{:keys [new-network-values network-bars amount->width reset-values-fn
+           lock-press-fn unlock-press-fn]}]
+  (lock-press-fn)
+  (js/setTimeout
+   (fn []
+     (align-bars-off-screen {:new-network-values new-network-values
+                             :network-bars       network-bars
+                             :amount->width      amount->width})
+     (reset-values-fn)
+     (js/setTimeout #(reset-bars-positions network-bars unlock-press-fn) 100))
```

Comment:
IMO we should store a reference and cancel timers before creating new ones, also cancel them if when component unmounts

Reply:
They are very short living, but I agree :thinking:  maybe we could create a common way to clean and set timeouts.

I've just added a mechanisms to track timeouts for this component and also to clean them once they have been used or just before we try to relaunch them.

Could you please check it and provide feedback?
thanks! :)